test/assembler/parser.test.ts

```diff
@@ -1,158 +1,264 @@
-import { AreaType, ICodeFile, IInstruction } from 'assembler/ast'
-import { createInstruction, parse, removeNonCode } from 'assembler/parser'
+import { AreaType } from 'assembler/ast'
+import { parse } from 'assembler/parser'
+import { ITextParseRule, parseText } from 'assembler/parser/text'
 
-//// create test data
+const code = `
+MY_CONSTANT EQU 0x123
+AREA |.data|, DATA, READWRITE
+  DCD 0xFF ; A comment
```

Comment:
I am missing tests for comments on own line and empty lines without anything on it

Reply:
I extended the tests.